### ISSUE TITLE
consult-org-heading can feel like wading through molasses

### ISSUE THREAD

**Issue by tecosaur** [minad/consult#708]
Heyo,

I'm a fan of `consult-org-heading` as a wonderfully effective way to jump around. Unfortunately, I find it nigh-unusable on my config. After attempting `consult-org-heading` twice using ELP, I recorded 15 invocations of `org-fold-core-get-regions` which cause `org-fold-core-next-folding-state-change` to be called 20,000 times — which takes 18 seconds.

In #563 the idea of a transactional API for org-fold was raised
> Yes, as we discussed, such a transactional API would be nice to have but
is probably not necessary for now. If performance issues are observed in
the future, I will revisit this.

I imagine this might count as a motivating example for a transactional API.

In any case, it would be nice to be able to reasonably use `consult-org-heading` on my config again.

cc: @yantar92

**Comment by minad** [comment]
Anyway, I don't think we can do anything from the side of Consult. Consult is designed to work with a transactional API - therefore I made this proposal in the discussion with Ihor. As soon as Org provides such an API, we can use it here and profit from the performance improvements. It is unlikely that I will find time to work myself on a patch for Org. I just took on the new responsibility of maintaining the [Compat](https://github.com/emacs-compat/compat) library. Btw. Compat could be interesting for Org itself. For your use case, I recommend for now to disable automatic preview for `consult-org-heading` or to configure a delay via `consult-customize`.
